providers/okta/okta.go

```diff
@@ -92,6 +94,60 @@ func (c *Client) GroupByName(ctx context.Context, groupName string) (*okta.Group
 	return nil, fmt.Errorf("unable to find okta group %q", groupName)
 }
 
+func (c *Client) GroupsByName(ctx context.Context, groupNames []string, batchsize int) ([]*okta.Group, error) {
+	groups := []*okta.Group{}
+
+	batches := buildFilterNameBatches(groupNames, batchsize)
+
+	for _, filter := range batches {
+		// expanding stats to get the number of users in the group
+		query := c.GroupAPI.ListGroups(ctx).Search(filter).Expand("stats")
+
+		oktaGroups, resp, err := query.Execute()
+
+		for {
+			if err != nil {
+				bdy, err2 := io.ReadAll(resp.Body)
+				if err2 != nil {
+					return nil, fmt.Errorf("failed to read response body: %s %w", string(bdy), err)
+				}
+
+				return nil, fmt.Errorf("failed to query okta group: %w", err)
```

Comment:
nit: to differentiate from `GroupByName` failure.
```suggestion
				return nil, fmt.Errorf("failed to query okta groups: %w", err)
```